Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialAsset.cpp

```diff
@@ -182,6 +194,52 @@ namespace AZ
             m_isDirty = false;
         } 
 
+        template <typename iteratorType>
+        void MaterialAsset::RenamePropertyNames(iteratorType start, iteratorType end)
+        {
+            for (iteratorType it = start; it != end; ++it)
+            {
+                for (auto& propertyName : m_propertyNames)
+                {
+                    const auto toFromIterator = it->find(propertyName.GetCStr());
+                    if (toFromIterator != it->end())
+                    {
+                        propertyName = AZ::Name{ toFromIterator->second };
+                    }
+                }
+            }
+        }
+
+        void MaterialAsset::RenamePropertyNames()
+        {
+            // construct toFrom map in ascending order of version.
+            AZStd::vector<AZStd::map<AZStd::string, AZStd::string>> versionToFrom;
+
+            const bool ascending = m_materialTypeVersion < m_materialTypeAsset->GetVersion();
+            for (int i = 0; i < AZStd::abs(static_cast<int>(m_materialTypeVersion - m_materialTypeAsset->GetVersion())); ++i)
+            {
+                const auto& versionUpdate = m_materialTypeAsset->GetMaterialVersionUpdate((m_materialTypeVersion < m_materialTypeAsset->GetVersion() ? m_materialTypeVersion : m_materialTypeAsset->GetVersion()) + i + 1);
+
+                versionToFrom.push_back();
+                for (const auto& action : versionUpdate.m_actions)
+                {
+                    const AZStd::string from = ascending ? action.m_argsMap.find("from")->second : action.m_argsMap.find("to")->second;
+                    const AZStd::string to = ascending ? action.m_argsMap.find("to")->second : action.m_argsMap.find("from")->second;
+
+                    versionToFrom[i][from] = to;
```

Comment:
After switching to support only ascending, you can probably just apply the rename inline here, instead of building up a versionToFrom from.